ctapipe/calib/camera/calibrator.py

```diff
@@ -109,9 +109,21 @@ def _calibrate_dl0(self, event, telid):
     def _calibrate_dl1(self, event, telid):
         waveforms = event.dl0.tel[telid].waveform
         selected_gain_channel = event.r1.tel[telid].selected_gain_channel
+        dl1_calib = event.calibration.tel[telid].dl1
+
         if self._check_dl0_empty(waveforms):
             return
+
         n_pixels, n_samples = waveforms.shape
+
+        # subtract any remaining pedestal before extraction
+        if dl1_calib.pedestal_offset is not None:
+            # this copies intentionally, we don't want to modify the dl0 data
+            # waveforms have shape (n_pixel, n_samples), pedestals (n_pixels, )
+            # so we need to transpose to be able to utilize broadcasting
+            # and transpose back after to retain the correct shape
+            waveforms = (waveforms.T - dl1_calib.pedestal_offset).T
+
```

Comment:
Should we really silently ignore missing pedestals here, or issue a warning or exception if pedestals don't exist? If somebody wants to disable it, they should set the pedestal values to all 0, rather than None, I would guess (None means something failed in loading)

Reply:
For our default case of simtel array mcs, the pedestal is none, because it is already applied at the r1 stage.

So I don't think we should require setting it to 0 if this correction is not necessary.

Reply:
but I understood that only the dark pedestal was applied, and not the sky pedestal, or has that changed? If not, it would still be good for SimTelEventSource to write a "0" pedestal to the container, to be clear it is not needed.

Reply:
I don't think simtel differentiates these two pedestals... there is just one pedestal in the monitoring container.

Reply:
I would make these corrections optional and require that these are either None or an array of len(pixels).

Reply:
ok for now.